ports/raspberrypi/common-hal/wifi/Radio.c

```diff
@@ -81,7 +92,10 @@ mp_obj_t common_hal_wifi_radio_get_hostname(wifi_radio_obj_t *self) {
 }
 
 void common_hal_wifi_radio_set_hostname(wifi_radio_obj_t *self, const char *hostname) {
-    ro_attribute(MP_QSTR_hostname);
+    self->hostname = mp_obj_new_str(hostname, strlen(hostname));
+    hostname = mp_obj_str_get_str(self->hostname);
```

Comment:
This needs to be changed, as the storage for `self->hostname` will be freed when the VM exits. The storage needs to be statically allocated or otherwise allocated in a way that will survive a VM reset.

Reply:
In Espressif, the underlying ESP-IDF code keeps the hostname. Does the CYW43 allow it to be set and retrieved?

Reply:
No, lwip just takes a copy of the pointer value.
```
#define netif_set_hostname(netif, name) do { if((netif) != NULL) { (netif)->hostname = name; }}while(0)
```

Reply:
I'm confused about why we would want a (custom) hostname to survive a VM reset. Don't we want a clean slate?